docs/index.rst

```diff
@@ -6,6 +6,7 @@ Qiskit Terra documentation
   :maxdepth: 2
   :hidden:
 
+  How to <how_to/index>
```

Comment:
Unlike my concern in https://github.com/Qiskit/qiskit-terra/pull/8685/files#r1188677398 about adding a new top level section based on https://diataxis.fr/, this one seems fully justified. Why the difference? This has several guides included so it's not an ~empty section.

Reply:
You mean why here we have several guides but only one explanation? The short answer is because it's a lot faster and more straightforward to write a how-to guide.

For an explanation you need to think more about things like "why do we do things the way we do?" or "what is (insert important concept that most people don't really understand) about?". Then you need to think a lot more about how you present that information while a how-to only requires going though the steps of a specific task.